src_c/geometry.c

```diff
@@ -8,6 +8,108 @@
 
 #define PYGAMEAPI_GEOMETRY_NUMSLOTS 21
 
+static PyObject *
+pg_raycast(PyObject *_null, PyObject *const *args, Py_ssize_t nargs)
+{
+    double start_x, start_y;
+    double end_x, end_y;
+    PyObject **farr;
+    Py_ssize_t col_length;
+    Py_ssize_t loop;
+    double angle;
+    double max_dist;
+
+    if (nargs == 3) {
+        if (!PySequence_Check(args[0]) || !PySequence_Check(args[1]) || !PySequence_FAST_CHECK(args[2])) {
+            return RAISE(PyExc_TypeError,
+                        "argument of raycast() must be a sequence");
+        }
+        
+        if (!pg_TwoDoublesFromObj(args[0], &start_x, &start_y)) {
+            return RAISE(PyExc_TypeError, "the starting position requires a pair of floats");
+        }
+        if (!pg_TwoDoublesFromObj(args[1], &end_x, &end_y)) {
+            return RAISE(PyExc_TypeError, "the end position requires a pair of floats");
+        }
+        farr = PySequence_Fast_ITEMS(args[2]);
+        col_length = PySequence_Fast_GET_SIZE(args[2]);
+    }
+    else if (nargs == 4) {
+        if (!pg_TwoDoublesFromObj(args[0], &start_x, &start_y)) {
+            return RAISE(PyExc_TypeError, "the starting position requires a pair of floats");
+        }
+        if (!PyFloat_Check(args[1]) && !pg_DoubleFromObj(args[1], &angle)) {
+            return RAISE(PyExc_TypeError, "invalid angle");
+        }
+        if (!PyFloat_Check(args[2]) && !pg_DoubleFromObj(args[2], &max_dist)) {
+            return RAISE(PyExc_TypeError, "invalid max distance");
+        }
```

Comment:
Why should these check for PyFloat? There's no need as `pg_DoubleFromObj` already converts floats and ints to double anyway.

Reply:
Plus there no need to error if someone passes say 10 as angle or distance, we should consider 10 as 10.0 and that's it.